### ISSUE TITLE
Plugin API Request: IOrganizer methods to get lists of profiles, instances and applications.

### ISSUE THREAD

**Issue by Kezyma** [ModOrganizer2/modorganizer#1631]
**Is your feature request related to a problem? Please describe.**
A few of my plugins need to get the following lists and currently I work around them by trying to determine what the options are manually;
- A list of profiles for the current instance.
- A list of instances on this computer.
- A list of executables added to the instance.

**Describe the solution you'd like**
IOrganizer.getAllProfiles() to get a list of strings with the profile names.
IOrganizer.getAllInstances() to get a list of strings with the instance names.
IOrganizer.getAllApplications() to get a list of ExecutableInfo or something comparable with the information on added applications.

**Describe alternatives you've considered**
I just search and parse stuff manually to figure it out.

**Comment by Holt59** [comment]
The list of all profiles and the list of all applications is probably not too complicated to add, although `ExecutableInfo` is something specific to game plugins so that would probably be a different type of objects.

There is no way currently to list all instances on a computer, only global instances can be listed, even within MO2 itself.

**Comment by Kezyma** [comment]
> 

Yeah, executableinfo was just an example, any object with the name of the exe in MO, the exe path and the command line arguments would be good!

A list of global instances would be good, basically just the list of instances in Mod Organizer's instance selection menu!
